bgpd/bgp_route.c

```diff
@@ -11948,6 +11973,7 @@ DEFUN (bgp_distance,
 	bgp->distance_ebgp[afi][safi] = atoi(argv[idx_number]->arg);
 	bgp->distance_ibgp[afi][safi] = atoi(argv[idx_number_2]->arg);
 	bgp->distance_local[afi][safi] = atoi(argv[idx_number_3]->arg);
+	bgp_announce_routes_distance_update(bgp);
```

Comment:
Guess this will download/update all the routes to zebra again. one quick optimization that can be done is to call the announce API, only if any of the three distances actually changed i.e. modified from old value.